scripts/myMangaDex.js

```diff
@@ -128,144 +128,154 @@ class MyMangaDex {
-								value: "Update", type: "success",
-								onClick: (notification) => {
-									notification.close();
-									this.updateManga(usePepper, setStatus, true);
-								}
-							}, {
-								value: "Close", type: "error",
-								onClick: (notification) => {
-									notification.close();
-								}
-							}]
-						}, { position: "bottom-left", closeOnClick: false, sticky: true });
-					} else {
-						this.notification(NOTIFY.INFO, "Not updated", "The current chapter is not the next one and it wasn't updated on **MyAnimelist**.", this.getCurrentThumbnail());
-					}
-					return;
+			/*let isNext = (realChapter < 0 ||
+				//realChapter == this.manga.lastMyAnimeListChapter ||
+				realChapter == this.manga.lastMyAnimeListChapter + 1); // ||
+					//(!this.options.saveOnlyHigher && realChapter == this.manga.lastMyAnimeListChapter - 1));*/
+			const maybeNext = isHigherDex && realChapter <= Math.floor(this.manga.lastMangaDexChapter)+1;
+			if (this.options.saveOnlyNext && this.manga.lastMyAnimeListChapter > 0 && !maybeNext) {
+				if (this.options.confirmChapter) {
+					SimpleNotification.info({
+						title: "Not updated",
+						image: this.getCurrentThumbnail(),
+						text: "The current chapter is not the next one and it wasn't updated on **MyAnimelist**.\nYou can update it anyway or ignore this notification.",
+						buttons: [{
+							value: "Update", type: "success",
+							onClick: (notification) => {
+								notification.close();
+								this.updateManga(usePepper, setStatus, true);
+							}
+						}, {
+							value: "Close", type: "error",
+							onClick: (notification) => {
+								notification.close();
+							}
+						}]
+					}, { position: "bottom-left", closeOnClick: false, sticky: true });
+				} else {
+					this.notification(NOTIFY.INFO, "Not updated", "The current chapter is not the next one and it wasn't updated on **MyAnimelist**.", this.getCurrentThumbnail());
 				}
+				return;
+			}
+		}
 
-				let doUpdate = this.manga.lastMyAnimeListChapter != realChapter;
-				let oldStatus = this.manga.status;
-				if (this.mangaDexScore > 0) {
-					this.manga.score = this.mangaDexScore;
-				}
-				let { requestURL, body } = buildMyAnimeListBody(usePepper, this.manga, this.csrf, setStatus);
-
-				// Send the POST request to update the manga if there as a change
-				if (doUpdate) {
-					await browser.runtime.sendMessage({
-						action: "fetch",
-						url: requestURL,
-						options: {
-							method: "POST",
-							body: body,
-							redirect: "follow",
-							credentials: "include",
-							headers: {
-								"Content-Type": "application/x-www-form-urlencoded",
-								"accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"
-							}
-						}
-					});
+		const doUpdate = doMyAnimeList && this.manga.lastMyAnimeListChapter != realChapter;
+		const oldStatus = this.manga.status;
+		/*if (this.mangaDexScore > 0) {
+			this.manga.score = this.mangaDexScore;
+		}*/
```

Comment:
This always overrides the updated score from the modal. it's not possible for this block to be executed in any other situation I believe (mangaDexScore on other pages not set)
Maybe we can find a better place for it?

Reply:
I don't really know if I was really using the MangaDex score for anything.